static/js/UserHistoryPanel.jsx

```diff
@@ -145,62 +114,59 @@ UserHistoryPanel.propTypes = {
   menuOpen:            PropTypes.string.isRequired,
 };
 
-const LibraryUserHistoryPanelWrapper = ({menuOpen, toggleLanguage, openDisplaySettings, openNav, compare, toggleSignUpModal}) => {
-  return (
-    <UserHistoryPanel menuOpen={menuOpen}
-                      toggleLanguage={toggleLanguage}
-                      openDisplaySettings={openDisplaySettings}
-                      openNav={openNav}
-                      compare={compare}
-                      toggleSignUpModal={toggleSignUpModal}
-                      dataSource={'library'} />
-  )
-};
 
-const SheetsUserHistoryPanelWrapper = ({menuOpen, toggleLanguage, openDisplaySettings, openNav, compare, toggleSignUpModal}) => {
-  return (
-    <UserHistoryPanel menuOpen={menuOpen}
-                      toggleLanguage={toggleLanguage}
-                      openDisplaySettings={openDisplaySettings}
-                      openNav={openNav}
-                      compare={compare}
-                      toggleSignUpModal={toggleSignUpModal}
-                      dataSource={'sheets'} />
-  )
+const dedupeItems = (items, saved) => {
+  /*
+  Deduplicates consecutive items with the same book or sheet_id.  In 'saved' mode, we don't deduplicate.
+  :param items: list of UserHistory objects to deduplicate
+  :param saved: bool: True if the items are saved, False if not
+  :return: list of deduplicated items
+  */
+  if (saved) {
+    return items; // Don't deduplicate saved items
+  }
+  
+  const deduped = [];
+  let prev = {};
+  
+  for (const item of items) {
+    const key = item.is_sheet ? 'sheet_id' : 'book';
+    if (item[key] !== prev[key]) {
+      deduped.push(item);
+      prev = item;
```

Comment:
i don't know if for first ittem it can be that `item[key]` will be undefined, but if so it will be equal to `prev[key]`. that can solve it:

```suggestion
let prevKey;
for (const item of items) {
  const key = item.is_sheet ? 'sheet_id' : 'book';
  if (item[key] !== prevKey) {
    deduped.push(item);
    prevKey = item[key];
```
(that also saves some memory, for `prevKey` is lighter than the whole `prev`)